src/query/graphite/common/percentiles.go

```diff
@@ -79,12 +78,19 @@ func SafeSum(input []float64) (float64, int) {
 		if !math.IsNaN(v) {
 			sum += v
 		} else {
-			nans += 1
+			nans++
 		}
 	}
 	return sum, nans
 }
 
+// SafeAverage returns the average of the input slice the number of NaNs in the input.
+func SafeAverage(input []float64) (float64, int) {
+	sum, nans := SafeSum(input)
+	count := len(input) - nans
```

Comment:
can they all be NaNs?

Reply:
Hm good call.

Reply:
Seems like this is actually safe:
https://play.golang.org/p/MhMnzAReOd1